jina/drivers/evaluate.py

```diff
@@ -118,4 +119,57 @@ def _apply_all(self,
                 gt_content = pb2array(gt_content)
             evaluation.value = self.exec_fn(doc_content, gt_content)
             evaluation.op_name = f'{self.metric}-{self.exec.metric}'
-            evaluation.ref_id = groundtruth.id
\ No newline at end of file
+            evaluation.ref_id = groundtruth.id
+
+
+class GroundTruthKVIndexDriver(BaseIndexDriver):
+    """Serialize the documents/chunks in the request to key-value JSON pairs and write it using the executor
+    """
+    def __init__(self, id_tag: str = 'id', *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._traversal_paths = ('r', )
+        self.id_tag = id_tag
+
+    def _apply_all(self, docs: Iterable['jina_pb2.Document'], *args, **kwargs) -> None:
+        keys = [doc.tags[self.id_tag] for doc in docs]
+        values = [doc.SerializeToString() for doc in docs]
+        self.exec_fn(keys, values)
+
+
+class LoadGroundTruthDriver(BaseExecutableDriver):
```

Comment:
```suggestion
class LoadGroundTruthDriver(BaseExecutableDriver):
```

should inherit from KVSearchDriver

Reply:
Solved!